spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -67,11 +69,15 @@ public boolean caseSensitive() {
   }
 
   public boolean localityEnabled() {
-    if (table.io() instanceof HadoopFileIO) {
-      HadoopInputFile file = (HadoopInputFile) table.io().newInputFile(table.location());
-      String scheme = file.getFileSystem().getScheme();
-      boolean defaultValue = LOCALITY_WHITELIST_FS.contains(scheme);
-      return PropertyUtil.propertyAsBoolean(readOptions, SparkReadOptions.LOCALITY, defaultValue);
+    if (table.io() instanceof HadoopFileIO || table.io() instanceof ResolvingFileIO) {
```

Comment:
I am a bit concerned for people using `ResolvingFileIO` with this approach, previously we will only create an input file to check locality if it's `HadoopFileIO`, but now if the user is using `RevolvingFileIO` this operation will be done for every single file even if it is not a HadoopFileIO for the specific location.

I am wondering if we should make `ResolvingFileIO.implFromLocation` a public method so we can know the FileIO used for the location without the need to open input file.

Reply:
> I am wondering if we should make ResolvingFileIO.implFromLocation a public method so we can know the FileIO used for the location without the need to open input file.

+1 on this, will certainly help.